File: samples/app/src/main/java/com/sourcepoint/example_app/MainActivity.java
package com.sourcepoint.example_app;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;

import androidx.appcompat.app.AppCompatActivity;

import com.sourcepoint.cmplibrary.BuilderV6;
import com.sourcepoint.cmplibrary.ClientInteraction;
import com.sourcepoint.cmplibrary.gdpr.GDPRConsentLibClient;
import com.sourcepoint.example_app.core.DataProvider;
import com.sourcepoint.gdpr_cmplibrary.*;

import kotlin.Lazy;
import org.jetbrains.annotations.Nullable;

import static org.koin.java.KoinJavaComponent.inject;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "**MainActivity";

    final static int accountId = 22;
    final static int propertyId = 7639;
    final static String propertyName = "tcfv2.mobile.webview";
    final static String pmId = "122058";

    private ViewGroup mainViewGroup;

    private final Lazy<DataProvider> dataProvider = inject(DataProvider.class);

    private void showView(View view) {
        if(view.getParent() == null){
            view.setLayoutParams(new ViewGroup.LayoutParams(0, 0));
            view.getLayoutParams().height = ViewGroup.LayoutParams.MATCH_PARENT;
            view.getLayoutParams().width = ViewGroup.LayoutParams.MATCH_PARENT;
            view.bringToFront();
            view.requestLayout();
            mainViewGroup.addView(view);
        }
    }
    private void removeView(View view) {
        if(view.getParent() != null) mainViewGroup.removeView(view);
    }

    private GDPRConsentLib buildGDPRConsentLib() {
        return GDPRConsentLib.newBuilder(accountId, propertyName, propertyId, pmId,this)
                .setOnConsentUIReady(this::showView)
                .setOnAction(actionType  -> Log.i(TAG , "ActionType: " + actionType.toString()))
                .setOnConsentUIFinished(this::removeView)
                .setOnConsentReady(consent -> {
                    // at this point it's safe to initialise vendors
                    for (String line : consent.toString().split("\n"))
                        Log.i(TAG, line);
                })
                .setAuthId(dataProvider.getValue().getAuthId())
                .setOnError(error -> Log.e(TAG, "Something went wrong"))
                .build();
    }

    // GDPRConsentLibClient
    private GDPRConsentLibClient buildGDPRConsentLibV6() {
        GDPRConsentLibClient consentLibClient =  new BuilderV6()
                .setAccountId(accountId)
                .setContext(this)
                .setPropertyName(propertyName)
                .setPropertyId(propertyId)
                .setPmId(pmId)
                .setClientInteraction(new ClientInter())
//                .setAuthId(dataProvider.getValue().getAuthId())
                .build(GDPRConsentLibClient.class);

//        consentLibClient.setClientInteraction(new ClientInter());

        return consentLibClient;
    }

    @Override
    protected void onResume() {
        super.onResume();
        buildGDPRConsentLibV6().loadMessage(dataProvider.getValue().getAuthId());
//        buildGDPRConsentLib().run();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mainViewGroup = findViewById(android.R.id.content);
        findViewById(R.id.review_consents).setOnClickListener(_v ->
                buildGDPRConsentLibV6().loadPrivacyManager(dataProvider.getValue().getAuthId())
//                buildGDPRConsentLib().showPm()
        );
        findViewById(R.id.auth_id_activity).setOnClickListener(_v -> startActivity(new Intent(this, MainActivityAuthId.class)));
    }

    class ClientInter implements ClientInteraction {

        @Override
        public void onConsentUIFinishedCallback(@Nullable View v) {
            removeView(v);
        }

        @Override
        public void onConsentUIReadyCallback(View v) {
            showView(v);
        }

        @Override
        public void onConsentReadyCallback(GDPRUserConsent consent) {
            // at this point it's safe to initialise vendors
            for (String line : consent.toString().split("\n"))
                Log.i(TAG, line);
        }

        @Override
        public void onErrorCallback(ConsentLibException v) {
            Log.e(TAG, "Something went wrong");
        }

        @Override
        public void onActionCallback(ActionTypes actionTypes) {
            Log.i(TAG , "ActionType: " + actionTypes.toString());
        }
    }
}

File: cmplibraryV6/src/androidTest/java/com/sourcepoint/cmplibrary/MainActivity.kt
package com.sourcepoint.cmplibrary

import android.app.Activity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.RelativeLayout.LayoutParams
import android.widget.Toast
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.constraintlayout.widget.ConstraintSet
import com.sourcepoint.cmplibrary.legislation.gdpr.GDPRConsentLib
import com.sourcepoint.cmplibrary.legislation.gdpr.SpGDPRClient
import com.sourcepoint.gdpr_cmplibrary.ActionTypes
import com.sourcepoint.gdpr_cmplibrary.ConsentLibException
import com.sourcepoint.gdpr_cmplibrary.GDPRUserConsent

class MainActivity : Activity() {

    companion object {
        val TAG = "**MainActivity"
    }

    private val rlp: LayoutParams by lazy {
        LayoutParams(
            LayoutParams.MATCH_PARENT,
            LayoutParams.MATCH_PARENT
        )
    }

    private val campaign = Campaign(
        22,
        7639,
        "tcfv2.mobile.webview",
        "122058"
    )

    val gdprConsent: GDPRConsentLib by lazy {
        Builder()
            .setAccountId(campaign.accountId)
            .setPropertyName(campaign.propertyName)
            .setPropertyId(campaign.propertyId)
            .setPmId(campaign.pmId)
            .setContext(this)
            .setAuthId("authId")
            .build(GDPRConsentLib::class.java)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val components: Pair<ConstraintLayout, Button> = getComponents()
        setContentView(components.first, rlp)
        val button = components.second
        button.setOnClickListener { Toast.makeText(this, "test", Toast.LENGTH_SHORT).show() }
        gdprConsent.spGdprClient = GdprClient()
    }

    override fun onResume() {
        super.onResume()
    }

    inner class GdprClient : SpGDPRClient {
        override fun onConsentReady(consent: GDPRUserConsent?) {
            for (line in consent.toString().split("\n").toTypedArray()) Log.i(TAG, line)
        }

        override fun onConsentUIFinished(v: View) {
            gdprConsent.removeView(v)
        }

        override fun onConsentUIReady(v: View) {
            gdprConsent.showView(v)
        }

        override fun onError(error: ConsentLibException?) {
            Log.e(TAG, "Something went wrong")
        }

        override fun onAction(actionTypes: ActionTypes?) {
            Log.i(TAG, "ActionType: " + actionTypes.toString())
        }
    }

    fun Activity.getComponents(): Pair<ConstraintLayout, Button> {
        val cl = ConstraintLayout(this)
        cl.id = View.generateViewId()
        setContentView(cl, rlp)

        val button = Button(this)
        button.id = View.generateViewId()
        button.text = "pm"
        cl.addView(button)
        val set = ConstraintSet()
        set.clone(cl)
        set.connect(
            button.id, // the ID of the widget to be constrained
            ConstraintSet.START, // the side of the widget to constrain
            cl.id, // the id of the widget to constrain to
            ConstraintSet.START // the side of widget to constrain to
        )
        set.connect(button.id, ConstraintSet.END, cl.id, ConstraintSet.END)
        set.connect(button.id, ConstraintSet.TOP, cl.id, ConstraintSet.TOP)
        set.connect(button.id, ConstraintSet.BOTTOM, cl.id, ConstraintSet.BOTTOM)
        set.applyTo(cl)
        return Pair(cl, button)
    }
}
